Gems/ROS2/Code/Source/SdfAssetBuilder/SdfAssetBuilder.cpp

```diff
@@ -0,0 +1,225 @@
+            [[maybe_unused]] constexpr const char* SdfAssetBuilderName = "SdfAssetBuilder";
+            constexpr const char* SdfAssetBuilderJobKey = "Sdf Asset Builder";
+            constexpr const char* SdfAssetBuilderSupportedFileExtensionsRegistryKey = "/ROS/SdfAssetBuilder/SupportedFileTypeExtensions";
+        }
+
+    SdfAssetBuilder::SdfAssetBuilder()
+    {
+        AssetBuilderSDK::AssetBuilderDesc sdfAssetBuilderDescriptor;
+
+        sdfAssetBuilderDescriptor.m_name = SdfAssetBuilderJobKey;
+        sdfAssetBuilderDescriptor.m_version = 1; // bump this to rebuild all sdf files
+        sdfAssetBuilderDescriptor.m_busId = azrtti_typeid<SdfAssetBuilder>();
+        sdfAssetBuilderDescriptor.m_patterns = GetSupportedBuilderPatterns();
+
+        sdfAssetBuilderDescriptor.m_createJobFunction = [this](auto && request, auto && response) 
+            { 
+                return CreateJobs(request, response); 
+            };
+
+        sdfAssetBuilderDescriptor.m_processJobFunction = [this](auto && request, auto && response) 
+            { 
+                return ProcessJob(request, response); 
+            };
+
+        // Listen for asset builder notifications requesting jobs for any of the sdf source file types.
+        BusConnect(sdfAssetBuilderDescriptor.m_busId);
+
+        // Register this builder with the AssetBuilderSDK.
+        AssetBuilderSDK::AssetBuilderBus::Broadcast(
+            &AssetBuilderSDK::AssetBuilderBus::Handler::RegisterBuilderInformation, sdfAssetBuilderDescriptor);
+    }
+
+    SdfAssetBuilder::~SdfAssetBuilder()
+    {
+        // Stop listening for asset builder notifications.
+        BusDisconnect();
+
+        // The AssetBuilderSDK doesn't support deregistration, so there's nothing more to do here.
+    }
+
+    AZStd::vector<AssetBuilderSDK::AssetBuilderPattern> SdfAssetBuilder::GetSupportedBuilderPatterns()
+    {
+        AZStd::vector<AssetBuilderSDK::AssetBuilderPattern> patterns;
+
+        auto settingsRegistry = AZ::SettingsRegistry::Get();
+        if (settingsRegistry == nullptr)
+        {
+            AZ_Error(SdfAssetBuilderName, false, "Settings Registry not found, no sdf file type extensions enabled.");
+            return {};
+        }
+
+        // Visit each supported file type extension and create an asset builder wildcard pattern for it.
+        auto VisitFileTypeExtensions = [&settingsRegistry, &patterns]
+            (const AZ::SettingsRegistryInterface::VisitArgs& visitArgs)
+            {
+                if (AZ::SettingsRegistryInterface::FixedValueString value;
+                    settingsRegistry->Get(value, visitArgs.m_jsonKeyPath))
+                {
+                    // Support both '.sdf' and 'sdf' style entries in the setreg file for robustness.
+                    AZStd::string wildcardPattern = (value[0] == '.')
+                        ? AZStd::string::format("*%s", value.c_str())
+                        : AZStd::string::format("*.%s", value.c_str());
```

Comment:
To protect against an invalid access when the string is empty, the [AZStd::string::starts_with](https://en.cppreference.com/w/cpp/string/basic_string/starts_with) function should be used instead which protects the string being empty.
```suggestion
                    // Support both '.sdf' and 'sdf' style entries in the setreg file for robustness.
                    AZStd::string wildcardPattern = value.starts_with('.')
                        ? value
                        : AZStd::string::format("*.%s", value.c_str());
```

Reply:
Makes sense, though the proposed fix isn't correct because if the value starts with `.` we still need to prepend a `*`. And to your point, if the string is empty or only contains a `.`, we probably shouldn't create a wildcard pattern at all. I'll make this a bit more robust.